include/swift/Option/Options.td

```diff
@@ -894,6 +894,13 @@ def no_allocations : Flag<["-"], "no-allocations">,
   Flags<[FrontendOption, HelpHidden]>,
   HelpText<"Diagnose any code that needs to heap allocate (classes, closures, etc.)">;
 
+def enable_strings : Flag<["-"], "enable-strings">,
```

Comment:
Would `-enable-embedded-strings` work? It's not a big deal, but `-enable-strings` seems a bit overly generic to me at first glance, unless we can think of other scenarios where `String` would be restricted outside of `-embedded` mode (and we could always introduce a more generic option later).

Reply:
I don't think we want flags unique to embedded if we can avoid it.

Reply:
Strings are implicitly available by default, so unless you are in embedded mode this flag doesn't make sense. In this phrasing, how can it not be specific to embedded? In order to make it generic, I think it would have to be `-disable-strings`, which could theoretically work in any configuration (but it's unclear why you'd use it outside of embedded).